panels/UserPanel.php

```diff
@@ -83,19 +83,26 @@ public function save()
             ]);
         }
 
-        $attributes = array_keys(get_object_vars($data));
-        if ($data instanceof ActiveRecord) {
-            $attributes = array_keys($data->getAttributes());
+        $identityData = $this->identityData($identity);
 
-            $attributeValues = [];
-            foreach ($attributes as $attribute) {
-                $attributeValues[$attribute] =  $data->getAttribute($attribute);
+        // If the identity is a model, let it specify the attribute labels
+        if ($identity instanceof Model) {
+            $attributes = [];
+
+            foreach ($identityData as $attribute => &$value) {
```

Comment:
It looks super-weird. `$value` is not used anywhere but still passed by reference. In this case performance isn't important at all. It is internal debugger page and there usually aren't that many attributes. I'd use `array_keys` there.